### PR TITLE
Update urban-dictionary extension

### PR DESCRIPTION
**Pull request by zhenpewu**
## Description

- migrate to latest @raycast/api
- update urban dictionary icon, as it is not following the icon format, for the new icon, it refers to https://twitter.com/urbandictionary

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/urban-dictionary/src/index.tsx [hunk 1]
```diff
@@ -139,156 +101,69 @@ function ItemDetails({ term }: { term: string }) {
...
-    results: [],
-    isLoading: true,
-  })
-  const cancelRef = useRef<AbortController | null>(null)
+function useSearch<T extends Endpoint>(endpoint: T, initial = "") {
+  const params = new URLSearchParams({ term: initial });
+  return useFetch(`https://api.urbandictionary.com/v0/${endpoint}?${params.toString()}`, {
+    method: "GET",
+    parseResponse: async (response) => {
+      const json = (await response.json()) as UrbanResponseList<typeof endpoint>;
```
Comment (zhenpewu) [inline extensions/urban-dictionary/src/index.tsx:128]: if (!response.ok) {
        throw new Error(response.statusText);
      }

should move before converting to json.

@pernielsentikaer I am not sure how to pull your commit to my local fork, do u mind fixing it ?

Comment (pernielsentikaer) [reply extensions/urban-dictionary/src/index.tsx]: Should be better now, do you mind checking 🙂

Comment (zhenpewu) [reply extensions/urban-dictionary/src/index.tsx]: lgtm, thanks.

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**